src/builders.ts

```diff
@@ -160,18 +156,71 @@ export class BomBuilder {
       report: new ThrowReport(),
       cacheOptions: { skipIntegrityCheck: true }
     }
-    const gatherLicenseTexts = this.gatherLicenseTexts
-    return async function (pkg: Package): Promise<PackageInfo> {
+    return async function (pkg: Package): Promise<any> {
       const { packageFs, prefixPath, releaseFs } = await fetcher.fetch(pkg, fetcherOptions)
       try {
         const manifestPath = ppath.join(prefixPath, 'package.json')
-        const packageInfo: PackageInfo = {
-          manifest: JSON.parse(await packageFs.readFilePromise(manifestPath, 'utf8'))
+        return JSON.parse(await packageFs.readFilePromise(manifestPath, 'utf8'))
+      } finally {
+        if (releaseFs !== undefined) {
+          releaseFs()
         }
-        if (gatherLicenseTexts) {
-          packageInfo.licenseEvidence = makeLicenseEvidence(prefixPath, packageFs)
+      }
+    }
+  }
+
+  readonly #LICENSE_FILENAME_PATTERN = /^(?:UN)?LICEN[CS]E|.\.LICEN[CS]E$|^NOTICE$/i
+
+  private async makeLicenseEvidenceFetcher (project: Project): Promise<LicenseEvidenceFetcher> {
+    const fetcher = project.configuration.makeFetcher()
+    const fetcherOptions: FetchOptions = {
+      project,
+      fetcher,
+      cache: await Cache.find(project.configuration),
+      checksums: project.storedChecksums,
+      report: new ThrowReport(),
+      cacheOptions: { skipIntegrityCheck: true }
+    }
+    const LICENSE_FILENAME_PATTERN = this.#LICENSE_FILENAME_PATTERN
+    const logger = this.console
+    return async function * (pkg: Package): AsyncGenerator<License> {
+      const { packageFs, prefixPath, releaseFs } = await fetcher.fetch(pkg, fetcherOptions)
+      try {
+        let files
+        try {
+          // option `withFileTypes:true` is not supported and casues crashes
+          files = packageFs.readdirSync(prefixPath)
+        } catch (e) {
+          logger?.warn('collecting license evidence in', prefixPath, 'failed:', e)
+          return
+        }
+        for (const file of files) {
+          if (!LICENSE_FILENAME_PATTERN.test(file)) {
+            continue
+          }
+
+          const contentType = getMimeForTextFile(file)
+          if (contentType === undefined) {
+            continue
```

Comment:
Do not silently skip over files matching LICENSE_FILENAME_PATTERN.

Reply:
current implementation is a copy/past of existing art.
lets discuss improvements in a different issue/pullrequest (see https://github.com/AugustusKling/cyclonedx-node-yarn/pull/1#discussion_r1841854471)